rust/integration-testing/src/bin/arrow-json-integration-test.rs

```diff
@@ -489,7 +489,12 @@ fn array_from_json(
             Ok(Arc::new(array))
         }
         DataType::Dictionary(key_type, value_type) => {
-            let dict_id = field.dict_id();
+            let dict_id = field.dict_id().ok_or_else(|| {
```

Comment:
I apologise, I do remember this change, but I'm not sure of what happened; I presume while I was rebasing.

Reply:
No worries! I remember I sent you a rebased PR to your branch that was... complicated... so I'm not surprised :) Thanks for the merge!